### PR TITLE
[Enhancement] (nereids)implement showCreateCatalogCommand in nereids

### LINKED ISSUE #42740: [Enhancement] (nereids)implement showCreateCatalogCommand in nereids
showCreateCatalog command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showCreateCatalog command in nereids:
unsupportedshowCreateCatalogStatement
``` SHOW CREATE CATALOG name=identifier                                           ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42740

implement showCreateCatalogCommand in nereids

Regression test result:

Changed  Session  Variables:
VarName                                                                                |  CurrentValue                |  DefaultValue
-----------------------------------------------|---------------------|-------------
character_set_results                                                    |  NULL                                |  utf8mb4          
enable_parallel_result_sink                                        |  true                                |  false              
sql_mode                                                                              |  STRICT_TRANS_TABLES  |                          
variable_version                                                              |  0                                      |  300                  
experimental_enable_nereids_distribute_planner  |  false                              |  true                

2024-11-19 15:27:31.633 INFO [suite-thread-1] (Suite.groovy:1316) - Execute tag: cmd, sql: show create catalog es
2024-11-19 15:27:31.641 INFO [suite-thread-1] (SuiteContext.groovy:392) - Delete /root/doris/doris/regression-test/data/nereids_p0/show/test_show_create_catalog.out
2024-11-19 15:27:31.642 INFO [suite-thread-1] (SuiteContext.groovy:394) - Generate /root/doris/doris/regression-test/data/nereids_p0/show/test_show_create_catalog.out
2024-11-19 15:27:31.665 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: drop catalog if exists es
2024-11-19 15:27:31.675 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_show_create_catalog in /root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_catalog.groovy succeed
2024-11-19 15:27:52.891 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-19 15:29:23.320 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_catalog.groovy: group=p0,external,hive,external_docker,external_docker_hive, name=test_show_create_catalog
2024-11-19 15:29:23.321 INFO [main] (RegressionTest.groovy:492) - All suites success.

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateCatalogCommand.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.google.common.base.Strings;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents the command for SHOW CREATE CATALOG.
+ */
+public class ShowCreateCatalogCommand extends Command implements NoForward {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateCatalogCommand.java:42]:
```suggestion
public class ShowCreateCatalogCommand extends ShowCommand {
```

File: regression-test/suites/export_p0/test_show_create_catalog.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    if (enabled != null && enabled.equalsIgnoreCase("true")) {
+        String catalog_name = "es"
+
+        sql """drop catalog if exists ${catalog_name}"""
+        sql """create catalog if not exists ${catalog_name} properties (
+            "type"="es",
+            "hosts"="http://127.0.0.1:9200"
+        );"""
+
+        result = sql """show create catalog ${catalog_name}"""
```
Comment (starocean999) [inline regression-test/suites/export_p0/test_show_create_catalog.groovy:30]: checkNereidsExecute("""show create catalog ${catalog_name}""")

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateCatalogCommand.java (hunk 1 above)
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateCatalogCommand.java]: done - pls review

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**